### PR TITLE
Allowing to select what to extract from the subsetted Spectra

### PR DESCRIPTION
**Pull request by jorainer**
This PR adds code to enable to specify what to extract from the filtered `Spectra` object.

### CODE REVIEW

File: R/massql-parser.R
Comment (sgibb) [inline R/massql-parser.R:55]:
```suggestion
    if (anyNA(x) || !length(x))
```

**Review by sgibb — CHANGES_REQUESTED** [review]
Sorry, but IMHO you wrote too many functions :wink:

File: R/massql-parser.R
Comment (jorainer) [reply R/massql-parser.R:86]: OK. Fine for me.

Comment (jorainer) [reply R/massql-parser.R:86]: Actually, I will have to re-write this part. In the end we need to support: `*`, `MS1DATA`, `MS2DATA`, `scaninfo(MS1DATA)`, `scansum(MS1DATA)` etc. So, for now I will make a distinction for MS1DATA and MS2DATA with the additional functions (scaninfo etc) being supported later.

Comment (jorainer) [reply R/massql-parser.R]: I've now re-implemented the function to allow also later parsing of functions to extract data subsets (e.g. `scaninfo()` or `scansum()`).

File: NAMESPACE
Comment (sgibb) [inline NAMESPACE:14]: See `roxygen2` documentation for `.what_data` below.

File: NAMESPACE
Comment (sgibb) [inline NAMESPACE:16]: See `roxygen2` documentation for `.what_data` below.

File: NAMESPACE
Comment (sgibb) [inline NAMESPACE:19]: See `roxygen2` documentation for `.what_data` below.

File: NAMESPACE
Comment (sgibb) [inline NAMESPACE:24]: See `roxygen2` documentation for `.what_data` below.

**Review by sgibb — APPROVED** [review]
I am happy with your code but there is something weird with the `roxygen2` tags.

After fixing it you could merge it.

File: R/massql-parser.R
Comment (sgibb) [reply R/massql-parser.R]: Sorry, you see I don't know anything about the syntax of MassQL :smile: (just assuming its like SQL but recognized in an earlier PR it is not (e.g. combination of AND/OR))

Comment (jorainer) [reply R/massql-parser.R]: I know - I'm also still thinking in SQL terms. Not so sure if it was a good decision to have a query language similar, but still different than SQL...

**Comment by jorainer** [comment]
Thanks @sgibb for the review. I fixed now the roxygen issue - seems roxygen doesn't like if there is text/description next to `@import`. Waiting for review from @andreavicini before merging.

File: R/Spectra.R
Comment (andreavicini) [inline R/Spectra.R:30]: Maybe `MS2DATA`?

**Review by andreavicini — APPROVED** [review]
The changes seem very nice to me, thanks!

**Pull request merged**
